### PR TITLE
Cover additional PHP keyword edge cases

### PR DESCRIPTION
**Pull request by spawnia**
Resolves https://github.com/spawnia/sailor/issues/63

### CODE REVIEW

**Review by simPod — APPROVED** [review]

**Review by simPod — CHANGES_REQUESTED** [review]
I tried it in my project and we generate wrong 

```php
/**
 * @property array<int, A|B> $devices
 * @property string $__typename
 */
class QueryName extends \Spawnia\Sailor\ObjectLike
{
    /**
     * @param array<int, A|B> $xxx
     */
    public static function make($xxx): self
``` where `B` should be `_B`

**Comment by simPod** [comment]
But it is broken in the master as well. So we can merge this and I'll look into the issue.

File: examples/php-keywords/src/test.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php declare(strict_types=1);
+
+require __DIR__ . '/../vendor/autoload.php';
+
+$result = \Spawnia\Sailor\Simple\Operations\MyObjectQuery::execute();
```
[the remark this answers is not in the document]
Comment (spawnia) [reply examples/php-keywords/src/test.php]: no, fixed.

File: examples/php-keywords/test.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (spawnia) [reply examples/php-keywords/test.sh:1]: fixed.

File: Makefile [hunk 3]
```diff
@@ -35,17 +30,18 @@ infection: ## Runs mutation tests with infection
...
-approve: ## Accept the current generated code as expected
-	$(call approve_example,custom-types)
-	$(call approve_example,input)
-	$(call approve_example,polymorphic)
-	$(call approve_example,simple)
+approve: ## Generate code and approve it as expected
+	tests/generate-and-approve-examples.php
+
+.PHONY: examples
+examples: ## Test examples
```
Comment (simPod) [inline Makefile:38]:
```suggestion
test-examples:
```

**Review by simPod — APPROVED** [review]
Running this in prod rn with no issues.

**Comment by spawnia** [comment]
> Running this in prod rn with no issues.

That is the best kind of review 😉

How do you feel about cutting a stable release of Sailor? We are using it in a bunch of projects ourselves and have not run into major issues. Anything you would change before?

**Comment by simPod** [comment]
I'm slowly adopting it for more complex use-cases. Overall, was using it only for basic operations until recently on several projects.

I'd give it a bit more time to see whether I encounter other issues or not.

**Pull request merged**